### PR TITLE
Fix logout consent issue in sample react applications

### PR DESCRIPTION
**Pull request by chaminjay**
https://user-images.githubusercontent.com/90854808/159498340-07c437d8-1eae-4780-bb26-2ac5ca779f26.mov

### CODE REVIEW

File: samples/asgardeo-react-app/src/pages/home.tsx [hunk 1]
```diff
@@ -73,6 +75,18 @@ export const HomePage: FunctionComponent<HomePagePropsInterface> = (
...
+    *   handles the error occurs when the logout consent page is enabled
+    *   and the user clicks 'NO' at the logout consent page
+    */
+    useEffect(() => {
+        on(Hooks.SignOutFailed, (error) => {
+            if (error.description === "End User denied the logout request") {
+                trySignInSilently();
+            }
+        });
+    }, [on]);
```
Comment (thivi) [inline samples/asgardeo-react-app/src/pages/home.tsx:88]:
```suggestion
    }, [ on ]);
```

File: samples/asgardeo-react-ts-app/src/pages/landing.tsx [hunk 2]
```diff
@@ -33,6 +33,18 @@ const LandingPage: FunctionComponent<{}> = () => {
...
+    *   handles the error occurs when the logout consent page is enabled
+    *   and the user clicks 'NO' at the logout consent page
+    */
+    useEffect(() => {
+        on(Hooks.SignOutFailed, (error) => {
+            if (error.description === "End User denied the logout request") {
+                trySignInSilently();
+            }
+        });
+    }, [on]);
```
Comment (thivi) [inline samples/asgardeo-react-ts-app/src/pages/landing.tsx:46]:
```suggestion
    }, [ on ]);
```

**Review by thivi — APPROVED** [review]

**Review by thivi — APPROVED** [review]

File: samples/asgardeo-react-app/src/pages/home.tsx [hunk 3]
```diff
@@ -45,7 +45,9 @@ export const HomePage: FunctionComponent<HomePagePropsInterface> = (
         signOut,
         getBasicUserInfo,
         getIDToken,
-        getDecodedIDToken
+        getDecodedIDToken,
+        on,
+        trySignInSilently,
```
Comment (Wathsara) [inline samples/asgardeo-react-app/src/pages/home.tsx:50]: Any special reason to add `,` at the end?

File: samples/asgardeo-react-app/src/pages/home.tsx [hunk 4]
```diff
@@ -73,6 +75,18 @@ export const HomePage: FunctionComponent<HomePagePropsInterface> = (
         })();
     }, [ state.isAuthenticated ]);
 
+    /* 
+    *   handles the error occurs when the logout consent page is enabled
+    *   and the user clicks 'NO' at the logout consent page
+    */
```
Comment (Wathsara) [inline samples/asgardeo-react-app/src/pages/home.tsx:81]: Shall we format the comment?

File: samples/asgardeo-react-app/src/pages/home.tsx (hunk 3 above)
Comment (chaminjay) [reply samples/asgardeo-react-app/src/pages/home.tsx:50]: Trailing comma has been added by the IDE's formatter (and I'm used to add it too). 

> Trailing commas (sometimes called "final commas") can be useful when adding new elements, parameters, or properties to JavaScript code. If you want to add a new property, you can add a new line without modifying the previously last line if that line already uses a trailing comma. This makes version-control diffs cleaner and editing code might be less troublesome.


References:
- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Trailing_commas
- https://blog.logrocket.com/best-practices-using-trailing-commas-javascript/

Comment (brionmario) [reply samples/asgardeo-react-app/src/pages/home.tsx:50]: There is a lint rule to warn when there is are unnecessary commas ([comma-dangle rule](https://github.com/asgardeo/asgardeo-auth-react-sdk/blob/main/lib/.eslintrc.js#L55)). Check if your editor has properly configured the ESLint plugins.

```suggestion
        trySignInSilently
```

File: samples/asgardeo-react-ts-app/src/pages/landing.tsx [hunk 5]
```diff
@@ -33,6 +33,18 @@ const LandingPage: FunctionComponent<{}> = () => {
         }
     }, [ state.isAuthenticated, history ]);
 
+    /*
```
Comment (brionmario) [inline samples/asgardeo-react-ts-app/src/pages/landing.tsx:36]: Lets format these comments as well.

File: samples/asgardeo-react-app/src/pages/home.tsx (hunk 3 above)
Comment (chaminjay) [reply samples/asgardeo-react-app/src/pages/home.tsx:50]: Resolved by https://github.com/asgardeo/asgardeo-auth-react-sdk/pull/100

File: samples/asgardeo-react-ts-app/src/pages/landing.tsx (hunk 5 above)
Comment (chaminjay) [reply samples/asgardeo-react-ts-app/src/pages/landing.tsx:36]: Addressed by https://github.com/asgardeo/asgardeo-auth-react-sdk/pull/100

File: samples/asgardeo-react-app/src/pages/home.tsx (hunk 4 above)
Comment (chaminjay) [reply samples/asgardeo-react-app/src/pages/home.tsx:81]: Fixes added with https://github.com/asgardeo/asgardeo-auth-react-sdk/pull/102

**Pull request merged**